src/main/java/onboarding/Problem4.java

```diff
@@ -1,8 +1,72 @@
 package onboarding;
 
 public class Problem4 {
+
     public static String solution(String word) {
-        String answer = "";
-        return answer;
+        return new Solver4(word).solve();
+    }
+}
+
+final class Solver4 {
+
+    private final String word;
+    private int curIdx;
+    private char cur;
+
+    public Solver4(final String word) {
+        this.word = word;
+        this.curIdx = 0;
+        this.cur = this.word.charAt(0);
+    }
+
+    public String solve() {
+        StringBuilder builder = new StringBuilder();
+        while (!this.reachedEndOfWord()) {
+            try {
+                this.assertCharAlphabetical();
+                builder.append(this.reverseOfCur());
+            } catch (Exception e) {
```

Comment:
```suggestion
            } catch (IllegalArgumentException e) {
```
이렇게 확실히 어떤 예외를 catch 할건지 보여주는게 좋을 것 같습니다!